pkg/limayaml/validate.go

```diff
@@ -119,12 +119,22 @@ func Validate(y *LimaYAML, warn bool) error {
 		}
 	}
 
+	// TODO: This check should be removed when we completely eliminate `CPUType` from limayaml.
+	if len(y.CPUType) > 0 && warn {
+		logrus.Warn("The top-level `cpuType` field is deprecated and will be removed in a future release. Please migrate to `vmOpts.qemu.cpuType`.")
+	}
 	for arch := range y.CPUType {
 		if !slices.Contains(ArchTypes, arch) {
 			return fmt.Errorf("field `cpuType` uses unsupported arch %q", arch)
 		}
 	}
 
+	for arch := range y.VMOpts.QEMU.CPUType {
+		if !slices.Contains(ArchTypes, arch) {
+			return fmt.Errorf("field `vmOpts.qemu.cpuType` uses unsupported arch %q", arch)
+		}
+	}
```

Comment:
Probably `defaults.go` should just replicate the top-level CPUType to VMOpts (with a warning when they conflict), then no need to validate the CPUTypes here twice?

Reply:
The validation should happen in `pkg/qemu`.
`pkg/limayaml` shouldn't be aware of the archs implemented by the QEMU driver